### PR TITLE
#184581170 Setup Docker for Backend Application

### PR DESCRIPTION
**Pull request by ericndungutse**
#### What does this PR do?
Have docker setup for backend API

#### Description of Task to be completed?
Have the following functionality configured

- Running express server using docker compose
- Creating  and running postgres using docker compose

#### How should this be manually tested?
First make sure you have docker installed. In your terminal;

1. Set environment variables. NB: **Set DB_HOST to `db`**
2. Run `docker compose up -d db` to run postgres container
3. Run `docker compose logs` to check if database is ready for connection.
4. Run `docker compose build` to build Docker image.
5. Run `docker compose up app`
6. Run `docker compose exec -it app npx sequelize-cli db:migrate` to create tables
7. Use Postman or any other api testing app to make the following request on 127.0.0.01:3000;
   - `POST /user/create` Body request {userName, email, password}
   - `GET /user/all`

#### What are the relevant pivotal tracker/Trello stories?
Story ID: [#184581170 ](https://www.pivotaltracker.com/story/show/184581170)

### CODE REVIEW

**Review by wilbrord2 — APPROVED** [review]
good

**Review by 1MuhireDavid — APPROVED** [review]

**Review by AbdulKhaliq59 — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    build: .
+    container_name: amigos_ecom_app_c
+    image: app_amigos_ecom
+    ports:
+      - 3000:3000
+    environment:
+      - DB_NAME=amigos_ecom
+      - DB_USERNAME=postgres
+      - DB_USER_PASSWORD=test1234
+      - DB_HOST=db
```
Comment (Habinezajanvier) [inline docker-compose.yml]: Can we pick those data from `.env` rather than hard-coding them in `docker-compose` file

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  db:
+    container_name: amigos_ecom_db_c
+    image: postgres:12
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_DB=amigos_ecom
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=test1234
```
Comment (Habinezajanvier) [inline docker-compose.yml]: Sensible data should be in `.env`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    ports:
+      - 3000:3000
+    environment:
+      - DB_NAME=amigos_ecom
+      - DB_USERNAME=postgres
+      - DB_USER_PASSWORD=test1234
+      - DB_HOST=db
+    depends_on:
+      - db
+
```
Comment (Habinezajanvier) [inline docker-compose.yml]: find a way to add volumes tag on app services so that editing in dev mode can be synced in real time with changes in docker

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
Please squash your commits into one

**Review by nshutijonathan — APPROVED** [review]
LGTM

File: .env.example [hunk 4]
```diff
@@ -9,7 +9,11 @@ DB_USER=postgres
 DB_PASSWORD=YDB_PASSWORD
 DB_PORT=5432
 PORT=4000
+<<<<<<< HEAD
 COVERALLS_REPO_TOKEN=b5dAPmMgZHCARyJFm6QYaOXghAF7rTD9g
+=======
+
+>>>>>>> 99a2eca... Revert "chore(docker compose): set up docker compose file"
```
Comment (Habinezajanvier) [inline .env.example]: You didn't finish the rebase

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

File: .env.example (hunk 4 above)
Comment (ericndungutse) [reply .env.example]: Finished.

**Review by Habinezajanvier — APPROVED** [review]
🥇 LGTM

**Pull request merged**